### PR TITLE
Support Universal Assisted Model

### PR DESCRIPTION
**Pull request by cheungdaven**
## Description
[Provide a brief description of the changes in this PR]
PR to support universal assisted model, see https://huggingface.co/docs/transformers/main/en/generation_strategies#universal-assisted-decoding for more details.
## Related Issue
[Link to the related issue, if applicable]

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Other (please specify)

## Checklist
- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [ ] My code follows the project's coding style
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have updated the documentation accordingly
- [ ] I have added appropriate comments to my code, particularly in hard-to-understand areas

## Additional Information
[Any additional information, configuration, or data that might be necessary to reproduce the issue or test the PR]

## Screenshots (if appropriate)
[Add screenshots to help explain your changes]

## Performance Impact
[Describe any performance impact this change may have]

## Backwards Compatibility
[Describe any backwards incompatible changes and how you've addressed them]

## Dependency Changes
[List any new dependencies introduced or removed]

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by shreyash2106 — APPROVED** [review]
LGTM! 

Sorry for the delayed review.

File: src/agrag/modules/generator/generators/hf_generator.py [hunk 1]
```diff
@@ -80,10 +80,17 @@ def __init__(
...
         logger.info(f"Using Huggingface Model {self.assistant_model_name} as the Assistant Model")
         if self.assistant_model_name:
             assistant_model = AutoModelForCausalLM.from_pretrained(self.assistant_model_name).to(self.device)
             self.hf_generate_params["assistant_model"] = assistant_model
+            # To support Universal Assisted Decoding
+            # See https://huggingface.co/docs/transformers/main/en/generation_strategies#universal-assisted-decoding
+            if self.assistant_tokenizer_name:
+                assistant_tokenizer = AutoTokenizer.from_pretrained(self.assistant_tokenizer_name)
+                self.hf_generate_params["assistant_tokenizer"] = assistant_tokenizer
+                self.hf_generate_params["tokenizer"] = self.tokenizer
```
Comment (suzhoum) [inline src/agrag/modules/generator/generators/hf_generator.py:93]: It might be more logical to move this line after Line 81?

Comment (cheungdaven) [reply src/agrag/modules/generator/generators/hf_generator.py:93]: this line is usually not needed, but it is compulsory if there is an assistant tokenizer.

Comment (suzhoum) [reply src/agrag/modules/generator/generators/hf_generator.py:93]: OK that makes sense.

**Pull request merged**